modules/quality/include/opencv2/quality/QualityGMSD.hpp

```diff
@@ -0,0 +1,82 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef OPENCV_QUALITY_QUALITYGMSD_HPP
+#define OPENCV_QUALITY_QUALITYGMSD_HPP
+
+#include "QualityBase.hpp"
+
+namespace cv {
+namespace quality {
+namespace detail {
+namespace gmsd {
+
+    using mat_type = UMat;
+
+    // holds computed values for an input mat
+    struct mat_data
```

Comment:
1. Exposing internal implementation details in public API headers.

2.
> mat_data

Refer to OpenCV naming policy: https://github.com/opencv/opencv/wiki/Coding_Style_Guide

Reply:
@alalek  OK I will move it to a private class instead